activejob/lib/active_job/test_helper.rb

```diff
@@ -119,13 +119,19 @@ def queue_adapter_for_test
     #   end
     def assert_enqueued_jobs(number, only: nil, except: nil, queue: nil, &block)
       if block_given?
-        original_count = enqueued_jobs_with(only: only, except: except, queue: queue)
+        original_jobs = []
```

Comment:
It is not clear to me why we are building those two new array. We are only using the to count. Why is that? It seems there is too much side effects inside `enqueued_jobs_with` is the reason why we need new arrays is because of those side effects.

Reply:
I changed a bit the methods so that we don't use `count` directly in `jobs_with`, but `select` which builds the Array directly which removes the need to collect jobs here.